### PR TITLE
[TECH] rewrite client in typescript

### PR DESCRIPTION
**Pull request by Kroustille**

### CODE REVIEW

File: lib/api.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+var BASE_API = 'https://api.aircall.io/v1'
+
+var request = require('superagent')
```
Comment (tdebarochez) [inline lib/api.ts]:
```suggestion
import request from 'superagent'

const BASE_API = 'https://api.aircall.io/v1'
```

File: lib/numbers.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import API, { Options } from './api';
+
+var defaults = require('defaults');
```
Comment (tdebarochez) [inline lib/numbers.ts]:
```suggestion
import API, { Options } from './api';

import defaults from 'defaults';
```

**Review by tdebarochez — APPROVED** [review]
I didn't look into details, but overall LGTM

File: lib/api.ts (hunk 1 above)
Comment (Kroustille) [reply lib/api.ts]: I tried first and add some issues with request typings but I found a good way to do it, thanks 😁

**Pull request merged**